deepsubtree.go

```diff
@@ -168,7 +292,158 @@ func (dst *DeepSubTree) recursiveSet(node *Node, key []byte, value []byte) (
 	default:
 		return nil, false, fmt.Errorf("inner node does not have key set correctly")
 	}
-	return node, updated, nil
+	if updated {
+		return node, updated, nil
+	}
+	err = node.calcHeightAndSize(dst.ImmutableTree)
+	if err != nil {
+		return nil, false, err
+	}
+	orphans := dst.prepareOrphansSlice()
+	node.persisted = false
+	newNode, err := dst.balance(node, &orphans)
+	node.persisted = true
```

Comment:
Should this be be after the `error` check?

Reply:
It's doesn't change code behavior, but you're correct, it's cleaner to do the error check right after the call :)